addons/mass_mailing/static/src/themes/theme_service.js

```diff
@@ -0,0 +1,131 @@
+import { children } from "@html_editor/utils/dom_traversal";
+import { parseHTML } from "@html_editor/utils/html";
+import { markup } from "@odoo/owl";
+import { registry } from "@web/core/registry";
+import { Deferred } from "@web/core/utils/concurrency";
+import { Reactive } from "@web/core/utils/reactive";
+import { renderToString } from "@web/core/utils/render";
+
+const DEFAULT_ASSET = "mass_mailing.email_designer_themes";
+
+function hasDataOption(element, attribute) {
+    attribute = "data-" + attribute;
+    return element.hasAttribute(attribute) && element.getAttribute(attribute) !== "false";
+}
+
+function getClassName(name) {
+    return name ? "o_" + name + "_theme" : "";
+}
+
+function getNameFromClass(className) {
+    const match = className.match(/^o_(.*)_theme$/);
+    return match ? match[1] : undefined;
+}
+
+export class ThemeModel extends Reactive {
+    constructor(services) {
+        super();
+        this.orm = services.orm;
+        this.loadedAssets = new Set();
+        this.loadedThemes = new Map();
+        this.loadingPromise = new Deferred();
+    }
+
+    computeThemesTemplates(asset, themesEl) {
+        for (const theme of children(themesEl)) {
+            this.preProcessImages(theme);
+            const themeOptions = {
+                className: getClassName(theme.dataset.name),
+                hideFromMobile: hasDataOption(theme, "hide-from-mobile"),
+                html: markup(theme.innerHTML.trim()),
+                imgPath: theme.dataset.img || "",
+                layoutStyles: theme.dataset.layoutStyles || "",
+                name: theme.dataset.name,
+                nowrap: hasDataOption(theme, "nowrap"),
+                title: theme.getAttribute("title") || "",
+            };
+            if (hasDataOption(theme, "images-info")) {
+                const imagesInfo = Object.assign(
+                    { all: {} },
+                    JSON.parse(theme.dataset.imagesInfo || "{}")
+                );
+                for (const [key, info] of Object.entries(imagesInfo)) {
+                    imagesInfo[key] = Object.assign(
+                        {
+                            module: "mass_mailing",
+                            format: "jpg",
+                        },
+                        imagesInfo.all,
+                        info
+                    );
+                }
+                themeOptions.getImageInfo = (filename) => imagesInfo[filename] || imagesInfo.all;
+            }
+            // Wrap the Theme `html` with a technical layout.
+            themeOptions.html = markup(renderToString("mass_mailing.ThemeLayout", themeOptions));
```

Comment:
Hello, just a quick check for markup calls, this one can simply be `renderToMarkup`.